src/vaadin-grid.html

```diff
@@ -457,10 +457,11 @@
       }
 
       /** @protected */
-      __itemsReceived() {
+      __flushPendingRecalculateColumnWidths() {
         if (this._recalculateColumnWidthOnceLoadingFinished
-          && !this._cache.isLoading()
-          && this.__hasRowsWithClientHeight()) {
+        && !this._cache.isLoading()
+        && this.__hasRowsWithClientHeight()
+            && (!this._debounceIncreasePool || !this._debounceIncreasePool.isActive())) {
```

Comment:
Adding a debugger to `_recalculateColumnWidths` shows that it's invoked too often during the test. The reason is that the check on this row is missing from `recalculateColumnWidths()` which gets directly invoked in the test. I'd suggest creating a separate `__isCreatingRows()` function if you're going to have the check in 2 places.

Drafted the following proposal, ptal:
```js
__isCreatingRows() {
        // iron-list's _increasePoolIfNeeded function may have scheduled a debouncer that will re-invoke _increasePoolIfNeeded (in vaadin-grid-scroller)
        const schedulingDebounceIncreasePool = this._debouncers && this._debouncers._increasePoolIfNeeded && this._debouncers._increasePoolIfNeeded.isActive();
        // vaadin-grid-scroller's overridden _increasePoolIfNeeded may have scheduled a _debounceIncreasePool debouncer
        const debouncingIncreasePoolActive = this._debounceIncreasePool && this._debounceIncreasePool.isActive();

        // If either of the conditions are true, we consider the grid is still in the process of creating new rows
        return schedulingDebounceIncreasePool || debouncingIncreasePoolActive;
}
```

Notice how in the proposal, the column width calculation is blocked also when iron-list may have scheduled a debouncer to re-invoke the function. This means that `_increasePoolIfNeeded` debounceer may actually be inactive. So you'll need to add another call to `__flushPendingRecalculateColumnWidths` in vaadin-grid-scroller when the function is invoked but `_debounceIncreasePool` may have not gotten activated. I wonder if another test is required?